llama_stack/providers/inline/agents/meta_reference/openai_responses.py

```diff
@@ -90,19 +138,44 @@ def __init__(
         self.tool_groups_api = tool_groups_api
         self.tool_runtime_api = tool_runtime_api
 
-    async def get_openai_response(
-        self,
-        id: str,
-    ) -> OpenAIResponseObject:
+    async def _get_previous_response_with_input(self, id: str) -> OpenAIResponsePreviousResponseWithInputItems:
         key = f"{OPENAI_RESPONSES_PREFIX}{id}"
         response_json = await self.persistence_store.get(key=key)
         if response_json is None:
             raise ValueError(f"OpenAI response with id '{id}' not found")
-        return OpenAIResponseObject.model_validate_json(response_json)
+        return OpenAIResponsePreviousResponseWithInputItems.model_validate_json(response_json)
+
+    async def _prepend_previous_response(
+        self, input: str | list[OpenAIResponseInput], previous_response_id: str | None = None
+    ):
+        if previous_response_id:
+            previous_response_with_input = await self._get_previous_response_with_input(previous_response_id)
+
+            # previous response input items
+            new_input_items = previous_response_with_input.input_items.data
+
+            # previous response output items
+            new_input_items.extend(previous_response_with_input.response.output)
+
+            # new input items from the current request
+            if isinstance(input, str):
```

Comment:
there could be other inputs -- namely the list? why are they not added?

Reply:
All this is doing is ensuring if the input is a string, it gets converted to a list of size 1. When the input is already a list, we just use it as-is. However, re-reading this I get your point, and will adjust this so that if it's a string we `new_input_items.append(...)` else we `new_input_items.extend(...)`. That will make what this is doing more clear I believe.

Reply:
I adjusted these lines to something that makes it more clear what's happening here - 
```
            if isinstance(input, str):
                new_input_items.append(OpenAIResponseMessage(content=input, role="user"))
            else:
                new_input_items.extend(input)
```

That is cleaner that what I had previously, and makes it much more clear that we're handling both sides of that `if` statement.